File: app/src/main/java/com/judas/katachi/core/api/interceptors/UserAgentInterceptor.java
package com.judas.katachi.core.api.interceptors;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class UserAgentInterceptor implements Interceptor {
    private static final String USER_AGENT_HEADER = "User-Agent";

    @NonNull
    private final String userAgent;

    public UserAgentInterceptor(@NonNull final String userAgent) {
        this.userAgent = userAgent;
    }

    @NonNull
    @Override
    public Response intercept(@NonNull final Chain chain) throws IOException {
        return chain.proceed(addUserAgent(chain.request()));
    }

    private Request addUserAgent(final Request original) {
        return original.newBuilder()
                .header(USER_AGENT_HEADER, userAgent)
                .method(original.method(), original.body())
                .build();
    }
}

File: app/src/main/kotlin/com/judas/katachi/feature/api/interceptor/UserAgentInterceptor.kt
package com.judas.katachi.feature.api.interceptor

import com.judas.katachi.BuildConfig.*
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class UserAgentInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return chain.proceed(addUserAgent(chain.request()))
    }

    private fun addUserAgent(original: Request): Request {
        return original.newBuilder()
            .header("User-Agent", "Android [$APPLICATION_ID] ($VERSION_NAME-$BUILD_TYPE)")
            .method(original.method, original.body)
            .build()
    }
}
